### PR TITLE
Feat: Mapleopoly & Mastery/Recipe updates

### PR DESCRIPTION
**Pull request by Zintixx**
## Mapleopoly
- Based off the last version (Kritias)
- Added `Mapleopoly` `MapleopolyHandler` `MapleopolyPacket`
- Changed Opcodes from `DICE_GAME` to `MAPLEOPOLY` for clarity
- Added `MapleopolyTiles` as a DB table for flexibility
- Added `MapleopolySeeding` to implement Kritias iteration of the board

## Events
- Created an `GameEvent` folder to store the various types of events that are used in game.
- Added `MapleopolyEvent` and `StringBoardEvent` in the folder. There's plenty of more events but these are the two that are implemented first. See the list of (currently discovered) event types in `GameEvent`
- -Implemented all these events in the DB. Each event type has it's own table to store the specific data, parameters, etc, while the controlling table is `Events`. With this, there is an active column to specify where an event should be sent to the player.
- Seeding the following two GameEventTypes:
	- MapleopolyEvent - this is for the trip prizes (completing one full trip of the board)
	- StringBoard - This was previously just hardcoded, but now moved over. This is simply just a marquee seen in the top left

## Mastery Updates
- Basically this is just streamlining what we currently had and adding missing `MasteryPacket`s
- Now creating all the Masterys upon character creation due to it being needed in `WriteCharacter`

### CODE REVIEW

File: MapleServer2/PacketHandlers/Game/MasteryHandler.cs [hunk 1]
```diff
@@ -79,19 +87,39 @@ private void HandleCraftItem(GameSession session, PacketReader packet)
...
+                    {
+                        session.Send(MasteryPacket.MasteryNotice((short) MasteryNotice.RequiredQuestIsNotCompleted));
+                        return;
+                    }
+                }
+            }
+
             // does the play have enough mesos for this recipe?
-            if (!session.Player.Wallet.Meso.Modify(-recipe.GetMesosRequired()))
+            if (!session.Player.Wallet.Meso.Modify(-recipe.RequireMeso))
```
Comment (AlanMorel) [inline MapleServer2/PacketHandlers/Game/MasteryHandler.cs:113]: should `RequireMeso` be instead `RequiredMeso`?

File: MapleServer2/Types/GameEvent/GameEvent.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        TrafficOptimizer,
+        PetComposeSale,
+        SaleRemakes,
+        SaleEnchant,
+        ItemTradeRestriction,
+        HotTime,
+        UGCMapContractSale,
+        UGCMapExtensionSale,
+        DungeonBonusReward
+
```
Comment (AlanMorel) [inline MapleServer2/Types/GameEvent/GameEvent.cs:30]: Can remove this line

File: MapleServer2/Types/GameEvent/MapleopolyEvent.cs [hunk 3]
```diff
@@ -0,0 +1,13 @@
+﻿namespace MapleServer2.Types
+{
+    public class MapleopolyEvent // Mapleopoly
```
Comment (AlanMorel) [inline MapleServer2/Types/GameEvent/MapleopolyEvent.cs:3]: Unneeded comment

**Review by AlanMorel — CHANGES_REQUESTED** [review]

File: MapleServer2/PacketHandlers/Game/MapleopolyHandler.cs [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+                    session.Player.Mapleopoly.TotalTileCount += tileToStart;
+                    break;
+                case MapleopolyTileType.Start:
+                    break;
+                default:
+                    Console.WriteLine("Unsupported tile");
+                    break;
+
+            }
+            ProcessTrip(session); // Check if player passed Start
```
Comment (AngeloTadeucci) [inline MapleServer2/PacketHandlers/Game/MapleopolyHandler.cs:140]:
```suggestion
                    break;
            }

            ProcessTrip(session); // Check if player passed Start
```

File: MapleServer2/Types/GameEvent/GameEvent.cs (hunk 2 above)
Comment (AngeloTadeucci) [inline MapleServer2/Types/GameEvent/GameEvent.cs:30]: empty line

**Review by AngeloTadeucci — APPROVED** [review]

File: MapleServer2/PacketHandlers/Game/MasteryHandler.cs (hunk 1 above)
Comment (Zintixx) [reply MapleServer2/PacketHandlers/Game/MasteryHandler.cs:113]: sure, I just kept it line with what was currently there in the RecipeMetadata. We currently have all requirements as "RequireMastery","RequireQuest","RequireItem" etc. should we change them all to Required?

**Review by AngeloTadeucci — APPROVED** [review]

File: MapleServer2/PacketHandlers/Game/MasteryHandler.cs (hunk 1 above)
Comment (AlanMorel) [reply MapleServer2/PacketHandlers/Game/MasteryHandler.cs:113]: Nah nevermind
